atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/async/DefaultCassandraAsyncKeyValueServiceFactory.java

```diff
@@ -88,10 +85,7 @@ public ExecutorService visit(CqlCapableConfig cqlCapableConfig) {
      * @return a new dynamic thread pool with a thread keep alive time of 1 minute
      */
     private static ExecutorService createThreadPool(int maxPoolSize) {
-        LinkedBlockingQueue<Runnable> workQueue = new LinkedBlockingQueue<>();
-        NamedThreadFactory threadFactory = new NamedThreadFactory("Atlas Cassandra Async KVS", false);
-
-        return PTExecutors.newThreadPoolExecutor(0, maxPoolSize, 1, TimeUnit.MINUTES, workQueue, threadFactory);
```

Comment:
This executor configuration is highly suspect. At a glance I believe it will only ever have a single thread unless we manage to fill the LinkedBlockingQueue with Integer.MAX_VALUE elements.